File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/New/SubSystems/Bromine/Claws.java
package org.firstinspires.ftc.teamcode.New.SubSystems.Bromine;

import com.qualcomm.robotcore.hardware.HardwareMap;
import com.qualcomm.robotcore.hardware.Servo;

import org.firstinspires.ftc.teamcode.New.SubSystems.Java.JavaSubsystems;


public class Claws implements JavaSubsystems {
    @Override
    public void update() {
        leftClaw.update();
        rightClaw.update();
    }

    public LeftClaw leftClaw;
    public RightClaw rightClaw;


    public Claws(HardwareMap hardwareMap) {
        leftClaw = new LeftClaw(hardwareMap);
        rightClaw = new RightClaw(hardwareMap);
    }

    public static class LeftClaw implements JavaSubsystems {

        @Override
        public void update() {
            switch (state) {
                case Closed:
                    leftClaw.setPosition(States.Closed.pos);
                    break;
                case Open:
                    leftClaw.setPosition(States.Open.pos);
                    break;
            }
        }

        public States state = States.Open;
        Servo leftClaw;

        public enum States {
            Closed(.9), Open(.5);

            States(double pos) {
                this.pos = pos;
            }

            final double pos;
        }

        public LeftClaw(HardwareMap hardwareMap) {
            leftClaw = hardwareMap.get(Servo.class, "leftClaw");
        }

    }
    public static class RightClaw implements JavaSubsystems {

        @Override
        public void update() {
            switch (state) {
                case Closed:
                    rightClaw.setPosition(States.Closed.pos);
                    break;
                case Open:
                    rightClaw.setPosition(States.Open.pos);
                    break;
            }
        }

        public States state = States.Open;
        Servo rightClaw;

        public enum States {
            Closed(.9), Open(.5);

            States(double pos) {
                this.pos = pos;
            }

            final double pos;
        }

        public RightClaw(HardwareMap hardwareMap) {
            rightClaw = hardwareMap.get(Servo.class, "rightClaw");
        }
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/New/SubSystems/Bromine/Claws.kt
package org.firstinspires.ftc.teamcode.New.SubSystems.Bromine

import com.qualcomm.robotcore.hardware.HardwareMap
import com.qualcomm.robotcore.hardware.Servo
import org.firstinspires.ftc.teamcode.New.SubSystems.Kotlin.SubSystems

class ClawsKt(hardwareMap: HardwareMap) : SubSystems {

    val leftClaw = LeftClaw(hardwareMap)
    val rightClaw = RightClaw(hardwareMap)

    override fun update() {
        leftClaw.update()
        rightClaw.update()
    }

    class LeftClaw(hardwareMap: HardwareMap) : SubSystems, ColorSensorClaw(hardwareMap) {

        enum class States(val servoPose: Double) {
            Open(0.5), Closed(1.0),AutomaticClose(1.0);
        }

        override var state = States.Open
        val servo: Servo = hardwareMap.get(Servo::class.java, "rightClaw")
        override fun update() {
            when (state) {
                States.Open -> servo.position = States.Open.servoPose
                States.Closed -> servo.position = States.Closed.servoPose
                States.AutomaticClose -> {
                    if(checkForRecognition() == Recognition.Left){
                        servo.position = States.Closed.servoPose
                    }
                }
            }
        }

    }

    class RightClaw(hardwareMap: HardwareMap) : SubSystems, ColorSensorClaw(hardwareMap) {

        enum class States(val servoPose: Double) {
            Open(0.5), Closed(1.0),AutomaticClose(1.0);
        }

        override var state = States.Open

        val servo: Servo = hardwareMap.get(Servo::class.java, "leftClaw")
        override fun update() {

            when (state) {
                States.Open -> servo.position = States.Open.servoPose
                States.Closed -> servo.position = States.Closed.servoPose
                States.AutomaticClose -> {
                    if(checkForRecognition() == Recognition.Right){
                        servo.position = States.Closed.servoPose
                    }
                }
            }

        }

    }

}

